### PR TITLE
feat: global network metrics dashboard

### PR DESCRIPTION
**Pull request by jgryffindor**
## Description

- Added a metrics module, _put/get/delete_ endpoints, and related entity to create, retrieve, and delete Prometheus queries in the db.
- Added a metric-details module and _get_ endpoints to retrieve current and series values by the name of the metric entities stored in db.
- Added Metric-details controller. This adds two endpoints /series and /current that require a name parameter and two optional query params to define the from/to fields of a Grafana datasources request. 
- Added Metric-details service. This generates a request to Grafana's API from a Prometheus query stored in the database. This returns the current value of the most recent point in time or a series value based on the from/to fields or defaulting to the last 5 minutes.
- Added link to metrics page in the toolbar.

**Setup requirements:**
The Metric dashboard depends on .env being updated with Grafana auth creds and Prometheus datasource ID. 
Also depends on Prometheus queries existing in the database, otherwise will return NaN. 

## Related Issue

[operations #84](https://github.com/liftedinit/operations/issues/84)

## Testing

Changes were tested locally on Lifted's QA network. 
Tested restarts without .env file updates, no effect on Talib's scheduler occurred. 
Tested metricStat and metricChart components with missing props, added error handling
Tested metrics page loading if Grafana connection is unavailable, added error handling

## Breaking Changes (if applicable)

This should introduce no breaking changes to the existing schema.
I did make changes to app.module to import the Metric module and metric entities 

## Screenshots (if applicable)
image

## Checklist:

- [ ] I have read and followed the CONTRIBUTING guidelines for this project.
- [ ] I have added or updated tests and they pass.
- [ ] I have added or updated documentation and it is accurate.
- [ ] I have noted any breaking changes in this module or downstream modules.

### CODE REVIEW

File: client/src/ui/metricChart.tsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+          name: label,
+          data: seriesData,
+        },
+      ],
+    };
+  }
+
+  return (
+    <>
+    {isLoading || isError ? (
```
Comment (stanleyjones) [inline client/src/ui/metricChart.tsx:119]: 🤞 no errors! That's going to be spinning for a while. 😆

File: client/src/utils/index.ts [hunk 2]
```diff
@@ -41,3 +41,29 @@ export function get(path: string, params = {}) {
...
+
+export function convertGbToTb(b: number): number {
+  let gbValue = (b / (1000));
+  return gbValue
+}
+
+export function convertTbToPb(b: number): number {
+  let gbValue = (b / (1000 * 1000 ));
+  return gbValue
+}
```
Comment (stanleyjones) [inline client/src/utils/index.ts:69]: Did you want to do something like this here? https://stackoverflow.com/questions/10420352/converting-file-size-in-bytes-to-human-readable-string

**Review by stanleyjones — APPROVED** [review]
This is great! What an epic PR.

File: client/src/utils/index.ts (hunk 2 above)
Comment (jgryffindor) [reply client/src/utils/index.ts:69]: Yes and no! So yes I need a way to convert from different base units, some are returned from Prometheus as Bytes/Kb thus far. I needed it in raw format without the string label for chart series data, but also need the stats to show a label. Would be great to improve it with this and have a single dynamic conversion function that outputs forms for both situations. Not sure about unit standardization out of Prometheus, need to investigate more. Thank you!

File: client/src/ui/metricChart.tsx (hunk 1 above)
Comment (jgryffindor) [reply client/src/ui/metricChart.tsx:119]: Yes lol This is hacky to handle situations when Grafana API is unavailable. Will improve this in the next releases!

**Pull request merged**